pkg/pxc/service.go

```diff
@@ -51,6 +51,13 @@ func NewServicePXC(cr *api.PerconaXtraDBCluster) *corev1.Service {
 		obj.ObjectMeta.Labels["app.kubernetes.io/component"] = appName
 		obj.ObjectMeta.Labels["app.kubernetes.io/managed-by"] = "percona-xtradb-cluster-operator"
 		obj.ObjectMeta.Labels["app.kubernetes.io/part-of"] = "percona-xtradb-cluster"
+
+		obj.Spec.Ports = append(
```

Comment:
i think we should guard mysqlx port definition with versions check `cr.CompareVersionWith`